src/query/storages/fuse/src/operations/read_partitions.rs

```diff
@@ -57,6 +59,18 @@ impl FuseTable {
     ) -> Result<(PartStatistics, Partitions)> {
         debug!("fuse table do read partitions, push downs:{:?}", push_downs);
 
+        if let Some(PushDownInfo {
```

Comment:
Is it possible for the optimizer to check and return an empty result when `where 1=2`, even though we know that no records meet this condition?

Reply:
There's no need to do this optimization. It's fast enough now (0.01s).

Reply:
It's not an optimization.
This PR only fixed the fuse engine, not the other engines we have(perhaps have in the future) :)

Reply:
mysql> select sum(number) from numbers(100000000000);
+--------------------+
| sum(number)        |
+--------------------+
| 932355974711512064 |
+--------------------+
1 row in set (3.33 sec)
Read 100000000000 rows, 745.06 GiB in 3.319 sec., 30.13 billion rows/sec., 224.47 GiB/sec.

mysql> select sum(number) from numbers(100000000000) where 1=2;
+-------------+
| sum(number) |
+-------------+
|        NULL |
+-------------+
1 row in set (2.40 sec)
Read 100000000000 rows, 745.06 GiB in 2.396 sec., 41.74 billion rows/sec., 310.97 GiB/sec.